.changeset/introduce_the_domains_linter_feature.md

```diff
@@ -0,0 +1,43 @@
+---
+cli: minor
+---
+
+# Introduce the `domains` linter feature
+
+The Biome linter now has a new way to opt-in rules, with a concept called `domains`.
+
+Domains can be seen as concepts shared by different rules.
+
+You can enable and disable multiple rules that belong to a domain. When you assign `true`, Biome will enable the rules, when you assign `false`, Biome will disable the rules.
+
+```json5
+// biome.jsonc
+{
+  "linter": {
+    "domains": {
+      "test": false, // rules around testing are disabled
```

Comment:
It would be *really* neat if our `biome init` script could make a best effort to detect what kind of folder structure and/or file name pattern is in use for tests. And based on that it could automatically populate the `overrides`. For example:

```jsonc
{
    "overrides": [{
        "include": ["**/*.test.js"],
        "linter": {
            "domains": {
                "test": true
            }
        }
    }]
}
```

Reply:
We can go even further, and make this part of the `RuleDomain` feature. A `RuleDomain` can specify a `Matcher`, and enable the rules if it path matches a certain glob. This logic is applied the same way as dependencies  are applied, which means that if the user tinker with configuration, it's not applied anymore